src/sagemaker/amazon/image_classification.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from sagemaker.amazon.amazon_estimator import AmazonS3AlgorithmEstimatorBase, registry
+from sagemaker.amazon.common import file_to_image_serializer, response_deserializer
+from sagemaker.amazon.hyperparameter import Hyperparameter as hp
+from sagemaker.amazon.validation import gt, isin, ge, le
+from sagemaker.predictor import RealTimePredictor
+from sagemaker.model import Model
+from sagemaker.session import Session
+
+
+class ImageClassification(AmazonS3AlgorithmEstimatorBase):
+    repo = 'image-classification:latest'
+
+    num_classes = hp('num_classes', (gt(1)), 'num_classes should be an integer greater-than 1', int)
+    num_training_samples = hp('num_training_samples', (gt(1)),
+                              'num_training_samples should be an integer greater-than 1', int)
+    use_pretrained_model = hp('use_pretrained_model', (isin(0, 1),),
+                              'use_pretrained_model should be in the set, [0,1]', int)
+    checkpoint_frequency = hp('checkpoint_frequency', (ge(1),),
+                              'checkpoint_frequency should be an integer greater-than 1', int)
+    num_layers = hp('num_layers', (isin(18, 34, 50, 101, 152, 200, 20, 32, 44, 56, 110),),
+                    'num_layers should be in the set [18, 34, 50, 101, 152, 200, 20, 32, 44, 56, 110]', int)
```

Comment:
just a suggestion but maybe putting these in ascending order would make it easier for users to reason about? Or is there a reason why they are seemingly in a random order?

Reply:
Again, I don't want to because this is a logic that is used in the docs also. There is a reason why it is in this order and users familiar with this algorithm will find this ordering comfortable.